aws/logs_monitoring/settings.py

```diff
@@ -189,12 +190,34 @@ def __init__(self, name, pattern, placeholder):
     connect_timeout=5, read_timeout=5, retries={"max_attempts": 2}
 )
 # DD API Key
+# Check if the DD_API_KEY_SECRET_ARN environment variable is set
 if "DD_API_KEY_SECRET_ARN" in os.environ:
     SECRET_ARN = os.environ["DD_API_KEY_SECRET_ARN"]
     logger.debug(f"Fetching the Datadog API key from SecretsManager: {SECRET_ARN}")
-    DD_API_KEY = boto3.client("secretsmanager", config=boto3_config).get_secret_value(
+
+    # Fetch the secret from Secrets Manager
+    secret_response = boto3.client("secretsmanager", config=boto3_config).get_secret_value(
         SecretId=SECRET_ARN
-    )["SecretString"]
+    )
+
+    # The secret could be either a plain string or a JSON object
+    secret_string = secret_response.get("SecretString")
+
+    try:
+        # Try to parse the secret as JSON
+        secret_json = json.loads(secret_string)
+
+        # If it's a JSON object, look for the 'datadogKey' field
+        if "DD_API_KEY" in secret_json:
+            DD_API_KEY = secret_json["datadogKey"]
```

Comment:
Is this a specific key to your usecase ?

Reply:
Whoops, I meant to use `DD_API_KEY` here. Thank you.

Reply:
Fixed this, can you take another look please and let me know what you think?

Reply:
Closing this out since we renamed this.